### DISCUSSION TITLE
Could not resolve outlook.office.com

### DISCUSSION THREAD

**Discussion by skreddy6673** [concourse/concourse#6460] [General]
I have a pipeline that sends messages to Micrsoft Teams channel. The pipeline fails intermediately with 
```
* Could not resolve host: outlook.office.com; Unknown error
* Closing connection 0
curl: (6) Could not resolve host: outlook.office.com; Unknown error
+ curl_errcode=6
``` 

Concourse version: v6.5.1
Helm chart: v12.0.1
Microsoft Teams Version: 1.3.00.30866

**Comment by skreddy6673** [comment]
```
$ kubectl get pods -o wide -n concourse-test
NAME                                READY   STATUS    RESTARTS   AGE   IP            NODE                           NOMINATED NODE   READINESS GATES
concourse-dev-web-67bbbd8b8-n2mdx   1/1     Running   0          22m   10.0.11.226   ip-10-164-9-203.ec2.internal   <none>           <none>
concourse-dev-worker-0              1/1     Running   0          21m   10.0.15.225   ip-10-164-5-113.ec2.internal   <none>           <none>
```
On a worker:
```
# nslookup outlook.office.com
Server:         10.96.0.10
Address:        10.96.0.10#53

# cat /etc/resolv.conf
nameserver 10.96.0.10
```
On the Task Container:
```
nslookup outlook.office.com
Server:         10.0.15.225
Address:        10.0.15.225#53

cat /etc/resolv.conf
nameserver 10.0.15.225
```

I've made these configurational changes on worker garden:
```
dnsProxyEnable: false
      dnsServers:
        - 10.96.0.10
```

Sometimes i use to get
```
# nslookup outlook.office.com
;; Truncated, retrying in TCP mode.
Server:         10.96.0.10
Address:        10.96.0.10#53
```

**Reply by taylorsilva** [reply]
I have no idea what could be happening here. It might be a k8s networking problem with whatever dns server you're using in the cluster. I would start by checking there the `kube-dns-*` pods (if they're named that in your cluster). The DNS pods are probably installed in your `kube-system` namespace. Check their logs for errors?

**Comment by skreddy6673** [comment]
@vito @taylorsilva @cirocosta 
Sorry guys to pull you in, any pointer would be great.

**Reply by taylorsilva** [reply]
FYI, Ciro doesn't work on Concourse anymore so he's probably not going to respond.

**Comment by skreddy6673** [comment]
Nothing abnormal found in Kube-dns-*, an update was pushed to Outlook and it solved the issue, It's still a mystery.
